horace_core/algorithms/private/gen_sqw_check_params.m

```diff
@@ -60,6 +60,19 @@
         mess='Number of spe data sets must be a positive integer >= 1';
         return
     end
+    % if the lattice has been provided as a single lattice although there
+    % are more than one files, replicate it to give one lattice per file
+    if nfile>numel(lattice)
+        if numel(lattice)==1
+            lattice = repmat(lattice,nfile,1);
+        else
+            error('HORACE:gen_sqw_check_parms:invalid_argument', ...
+                  ['more than one lattice has been provided, ', ...
+                   'but their number differs from the number of files']);
+        end
```

Comment:
```suggestion
        if ~isscalar(lattice)
            error('HORACE:gen_sqw_check_parms:invalid_argument', ...
                  ['more than one lattice has been provided, ', ...
                   'but their number differs from the number of files']);
        end
        lattice = repmat(lattice,nfile,1);
```

Reply:
I don't think isscalar is clearer than numel==1. there is a bit of merit in processing the error then going onto the assignment, but not enough to make it worth changing